### ISSUE TITLE
[v2.0.0-alpha.2] Remote relationships are not derived when the joining column has a custom GraphQL column name

### ISSUE THREAD

**Issue by esseswann** [hasura/graphql-engine#6626]
This was mentioned in #6625 but I think it deserves its own issue
We cannot seem to view the remote relationship field under any role other than admin. `HASURA_GRAPHQL_ENABLE_REMOTE_SCHEMA_PERMISSIONS` is off
With role
![image](https://user-images.githubusercontent.com/6066417/109841181-dad11080-7c59-11eb-8ece-9abb61ec179e.png)

Without role
![image](https://user-images.githubusercontent.com/6066417/109841271-ea505980-7c59-11eb-9547-19585213f289.png)

We tried 
- switching on the `HASURA_GRAPHQL_ENABLE_REMOTE_SCHEMA_PERMISSIONS ` and allowing all access to the remote schema but it did not help
- fresh install of `v2.0.0-alpha.2` since originally it was an update 

It does reproduce when using the role from JWT (not a header in console)

**Comment by gurza** [comment]
Unfortunately, this issue exists in v2.0.0-alpha.3.

**Comment by codingkarthik** [comment]
Hello Folks,

Thank you for reporting this issue, but unfortunately I'm not able to reproduce this problem. I tried with both setting the `X-Hasura-Role` as a header in the console and through JWT. However, I would like to summarize the behaviour of remote join fields when remote schema permisssions are enabled and disabled. The remote schema permissions are enabled/disabled via the `HASURA_GRAPHQL_ENABLE_REMOTE_SCHEMA_PERMISSIONS` env variable, by default it's set to `false`.

1. When remote schema permissions are enabled
-------------------------------------------------

A remote join field will **only** be exposed to a particular role if the role has corresponding remote schema permissions for the remote schema being used by the remote join, if the graphql-engine is not able to derive the remote relationship with the permissions configured for the role, it will **not** be exposed for the role. More about this at: https://hasura.io/docs/latest/graphql/core/api-reference/metadata-api/remote-schema-permissions.html#cases-when-the-remote-relationship-cannot-be-derived-are


2. When remote schema permissions are disabled
--------------------------------------------------

A remote join field should always be accessible.

If this doesn't help and you are still facing this issue, it will really help me out if you can share the following details with me to debug this issue further:

1. Are remote schema permissions enabled? You should be able to see this in the startup logs generated by the graphql-engine.
2. Definition of the remote join
3. Remote schema permission configured for the said role

Thanks

**Comment by esseswann** [comment]
Thanks for reply @codingkarthik 
>Are remote schema permissions enabled? You should be able to see this in the startup logs generated by the graphql-engine.

The `HASURA_GRAPHQL_ENABLE_REMOTE_SCHEMA_PERMISSIONS` is definitely off just checked it:
![image](https://user-images.githubusercontent.com/6066417/110084331-e3802e80-7da0-11eb-96ee-d799d2016b65.png)

>Definition of the remote join
```yaml
remote_relationships:
- definition:
    hasura_fields:
    - image_id
    remote_field:
      image:
        arguments:
          id: $image_id
    remote_schema: images
  name: image
  ```

>Remote schema permission configured for the said role
There is no remote schema permission since it's not enabled

**Comment by esseswann** [comment]
Here is a minimum reproducing example for you http://18.133.241.39:8080/console/api/api-explorer
admin-secret codingkarthik
I think we know why you couldn't reproduce, we had some permissions on the hasura table. They're reflected in the example

**Comment by codingkarthik** [comment]
@esseswann Thanks for the repro, but I see that remote schema permissions were enabled in that instance, which is why the remote join field was not accessible. Once I added the remote schema permission, the remote join field was accessible. 
Do check it out and let me know 😄

**Comment by esseswann** [comment]
Sorry, we removed it and seems that indeed works so we are unable to reproduce it on a fresh instance. It still reproduces on the actual instance though. Currently trying to extend the example so it reproduces

**Comment by esseswann** [comment]
@codingkarthik Can we send you the admin secret to the actual instance somewhere personally like on LinkedIn? We're unable to make a minimum repro so I think we'll have to show you our insides :)

**Comment by esseswann** [comment]
@codingkarthik we are unable to reproduce it maybe we can send you an admin secret to the actual system in LinkedIn or any other DM?

**Comment by codingkarthik** [comment]
@esseswann Sure, can you send the necessary info at karthikeyan@hasura.io?

**Comment by codingkarthik** [comment]
I was finally able to reproduce this problem, special shout-out to @esseswann for the helping me with the repro 🎉 !  

Steps to repro:
---------------

1. Create any remote relationship
2. Customize the name of the column used to create the remote relationship

I'm not sure, why this is happening yet but I will try to update here as soon as I know more about it.

**Comment by gurza** [comment]
Thanks for quick fix! It works in v2.0.0-alpha.4.
